src/ext/Util/ca/scaexec.cpp

```diff
@@ -715,35 +860,127 @@ static HRESULT RemoveUserInternal(
+                    // Pass the entire string if it is too short
+                    // to have a \\ prefix.
+                }
+            }
+            else
+            {
+                wz = wzDomain;
+            }
+        }
+
+        er = ::NetGroupDel(wz, wzName);
+        if (NERR_GroupNotFound == er)
+        {
+            er = NERR_Success;
+        }
+        ExitOnFailure(hr = HRESULT_FROM_WIN32(er), "failed to delete group: %ls", wzName);
+    }
+    else
+    {
+        //
+        // Remove the group from other groups
+        //
+        pwz = wzGroupCaData;
+        while (S_OK == (hr = WcaReadStringFromCaData(&pwz, &pwzGroup)))
+        {
+            hr = WcaReadStringFromCaData(&pwz, &pwzGroupDomain);
+
+            if (FAILED(hr))
+            {
+                WcaLogError(hr, "failed to get domain for group: %ls, continuing anyway.", pwzGroup);
+            }
+            else
+            {
+                hr = RemoveGroupFromGroup(wzName, wzDomain, pwzGroup, pwzGroupDomain);
+                if (FAILED(hr))
+                {
+                    WcaLogError(hr, "failed to remove group: %ls from group %ls, continuing anyway.", wzName, pwzGroup);
+                }
+            }
+        }
+
+        if (E_NOMOREITEMS == hr) // if there are no more items, all is well
+        {
+            hr = S_OK;
+        }
+
+        ExitOnFailure(hr, "failed to get next group from which to remove group:%ls", wzName);
+    }
+LExit:
+    if (pDomainControllerInfo)
+    {
+        ::NetApiBufferFree(static_cast<LPVOID>(pDomainControllerInfo));
+    }
+
+    return hr;
+}
+
+
+static void GetServerName(LPCWSTR pwzDomain, LPWSTR* ppwzServerName)
 {
     DWORD er = ERROR_SUCCESS;
     PDOMAIN_CONTROLLER_INFOW pDomainControllerInfo = NULL;
 
     if (pwzDomain && *pwzDomain)
     {
-        er = ::DsGetDcNameW(NULL, (LPCWSTR)pwzDomain, NULL, NULL, NULL, &pDomainControllerInfo);
+        er = ::DsGetDcNameW(NULL, pwzDomain, NULL, NULL, NULL, &pDomainControllerInfo);
         if (RPC_S_SERVER_UNAVAILABLE == er)
         {
             // MSDN says, if we get the above error code, try again with the "DS_FORCE_REDISCOVERY" flag
-            er = ::DsGetDcNameW(NULL, (LPCWSTR)pwzDomain, NULL, NULL, DS_FORCE_REDISCOVERY, &pDomainControllerInfo);
+            er = ::DsGetDcNameW(NULL, pwzDomain, NULL, NULL, DS_FORCE_REDISCOVERY, &pDomainControllerInfo);
         }
 
         if (ERROR_SUCCESS == er && pDomainControllerInfo->DomainControllerName)
         {
             // Skip the \\ prefix if present.
             if ('\\' == *pDomainControllerInfo->DomainControllerName && '\\' == *pDomainControllerInfo->DomainControllerName + 1)
             {
-                *ppwzServerName = pDomainControllerInfo->DomainControllerName + 2;
+                *ppwzServerName = _wcsdup(pDomainControllerInfo->DomainControllerName + 2);
```

Comment:
I suspect _wcsdup(...) is not the preferred function / macro to use.  But my quick google couldn't find a 'nicer' option.

Reply:
StrAllocString(dest, src, 0 [size of src]) seems like the winner